File: netlify/functions/process_workbook.py
import json
def process_workbook(wb) -> any:
    print(wb)
    return wb
def handler(event) -> dict:
    try:
        wb = json.loads(event['body'])
    except json.JSONDecodeError:
        return {
            'statusCode': 400,
            'body': json.dumps({'error': 'Invalid request: Expected JSON body'})
        }
    processed_wb = process_workbook(wb)
    print('Workbook JSON correctly processed')
    return {
        'statusCode': 200,
        'body': json.dumps(processed_wb)
    }

File: netlify/functions/processWorkbook.ts
import { Handler } from "@netlify/functions";
function processWorkbook(wb: any): any {
  return wb;
}
const handler: Handler = async (
  event: any,
): Promise<{
  statusCode: number;
  body: string;
}> => {
  let wb;
  try {
    wb = JSON.parse(event.body as string);
  } catch (error) {
    return {
      statusCode: 400,
      body: JSON.stringify({ error: "Invalid request: Expected JSON body" }),
    };
  }
  const processedWb = processWorkbook(wb);
  return {
    statusCode: 200,
    body: JSON.stringify(processedWb),
  };
};
export { handler };
